agent/agent.go

```diff
@@ -550,11 +550,14 @@ func (a *Agent) Publisher(ctx context.Context, subscriptionID string) (exec.LogP
 func (a *Agent) nodeDescriptionWithHostname(ctx context.Context, tlsInfo *api.NodeTLSInfo) (*api.NodeDescription, error) {
 	desc, err := a.config.Executor.Describe(ctx)
 
-	// Override hostname and TLS info
+	// Override hostname
 	if desc != nil {
-		if a.config.Hostname != "" && desc != nil {
+		if a.config.Hostname != "" {
 			desc.Hostname = a.config.Hostname
 		}
+		if desc.Resources != nil && a.config.GenericResources != nil {
+			desc.Resources.Generic = a.config.GenericResources
```

Comment:
I think generic resource should be collected when a node joins the cluster, or when a node is updated (with changes to generic resource). This would allow hotplug resources to be used without node remove/add.